go/chat/boxer.go

```diff
@@ -740,3 +828,22 @@ func hashSha256V1(data []byte) chat1.Hash {
 	sum := sha256.Sum256(data)
 	return sum[:]
 }
+
+func compareClientHeaders(h1 chat1.MessageClientHeader, h2 chat1.MessageClientHeader) error {
+	if !h1.Conv.Eq(h2.Conv) {
+		return fmt.Errorf("Conv mismatch in chat message headers: %#v %#v", h1.Conv, h2.Conv)
+	}
+	if h1.TlfName != h2.TlfName {
+		return fmt.Errorf("TlfName mismatch in chat message headers: %#v %#v", h1.TlfName, h2.TlfName)
+	}
+	if h1.TlfPublic != h2.TlfPublic {
+		return fmt.Errorf("TlfPublic mismatch in chat message headers: %#v %#v", h1.TlfPublic, h2.TlfPublic)
+	}
+	if h1.MessageType != h2.MessageType {
+		return fmt.Errorf("MessageType mismatch in chat message headers: %#v %#v", h1.MessageType, h2.MessageType)
+	}
+	// TODO: Supersedes and other fields currently don't match after
+	// decryption. The easiest thing might be to stop using the same data
+	// structure on both sides.
```

Comment:
I should add a couple things, as noted in https://github.com/keybase/client/pull/6070